compiler/rustc_codegen_llvm/src/gotoc/mod.rs

```diff
@@ -165,8 +165,13 @@ impl<'tcx> GotocCtx<'tcx> {
         self.set_current_fn(instance);
         self.ensure(&self.fname(), |ctx, fname| {
             let mir = ctx.mir();
-            Symbol::function(fname, ctx.fn_typ(), None, ctx.codegen_span2(&mir.span))
-                .with_pretty_name(&ctx.pretty_name_from_instance(instance))
+            Symbol::function(
+                fname,
+                &ctx.readable_instance_name(instance),
```

Comment:
I wonder if we should have a `ctx.readable_fname()` that comes from the current function. Tho actually, we should maybe just rethink the `fname()` name as well.